File: src/dataset/typescript/sieve_Exercism-typescript-FilipeCerejo/claude/ClaudeSonnet4_sieve_v2.ts
export function primes(limit: number): number[] {
    if (limit < 2) return [];
    if (limit === 2) return [2];
    
    const sieve = new Uint8Array(limit - 1).fill(1);
    const sqrtLimit = Math.sqrt(limit);
    
    for (let i = 0; i < sieve.length && (i + 2) <= sqrtLimit; i++) {
        if (sieve[i]) {
            const prime = i + 2;
            for (let j = prime * prime - 2; j < sieve.length; j += prime) {
                sieve[j] = 0;
            }
        }
    }
    
    const result: number[] = [];
    for (let i = 0; i < sieve.length; i++) {
        if (sieve[i]) {
            result.push(i + 2);
        }
    }
    
    return result;
}

File: src/dataset/cpp/sieve_Exercism_MaxyMoos/claude/ClaudeSonnet4_sieve_v2.cpp
#include "sieve.h"
#include <vector>

using namespace std;

vector<int> sieve::primes(int range)
{
    if (range < 2) return vector<int>();
    
    vector<bool> is_prime(range + 1, true);
    is_prime[0] = is_prime[1] = false;
    
    for (int i = 2; i * i <= range; i++) {
        if (is_prime[i]) {
            for (int j = i * i; j <= range; j += i) {
                is_prime[j] = false;
            }
        }
    }
    
    vector<int> result;
    result.reserve(range / (log(range) > 0 ? log(range) : 1));
    
    for (int i = 2; i <= range; i++) {
        if (is_prime[i]) {
            result.push_back(i);
        }
    }
    
    return result;
}